ext/mbstring/php_mbregex.c

```diff
@@ -1361,7 +1361,7 @@ _php_mb_regex_ereg_search_exec(INTERNAL_FUNCTION_PARAMETERS, int mode)
 	OnigUChar *str;
 	OnigSyntaxType *syntax;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|ss", &arg_pattern, &arg_pattern_len, &arg_options, &arg_options_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "|s!s!", &arg_pattern, &arg_pattern_len, &arg_options, &arg_options_len) == FAILURE) {
```

Comment:
I suspect this function is not well behaved if you specify options but not pattern.

Reply:
Uh, I fail to see the problem :/ The only thing I noticed is that if you specify the `$pattern` but not `$options` argument, then the `options` variable will be uninitialized when passed to `php_mbregex_compile_pattern`.

Reply:
What I had in mind is that it will simply have no effect. The options are only used if a pattern is specified. But I guess that's okay?